### PR TITLE
refactor: simplify `PathUtilityService`

### LINKED ISSUE #454: Simplify `PathUtilityService`
[`PathUtilityService`][1] was created when Component Detection was initially ported from .NET Framework to an early version of .NET Core. Many of the file and path utilities we needed were not natively available in `System.IO`. However, in .NET 6 [`System.IO.Path`][2], [`System.IO.File`][3], and other [`System.IO` classes][4] are extremely comprehensive, and we should migrate to those instead.

[1]: https://github.com/microsoft/component-detection/blob/main/src/Microsoft.ComponentDetection.Common/PathUtilityService.cs
[2]: https://learn.microsoft.com/en-us/dotnet/api/system.io.path?view=net-6.0
[3]: https://learn.microsoft.com/en-us/dotnet/api/system.io.file?view=net-6.0
[4]: https://learn.microsoft.com/en-us/dotnet/api/system.io?view=net-7.0

### PR DESCRIPTION
**Pull request by melotic**
Closes #454

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Common/PathUtilityService.cs [hunk 1]
```diff
@@ -117,176 +53,14 @@ public bool MatchesPattern(string searchPattern, string fileName)
...
-        {
-            this.logger.LogWarning("Detected execution in a Windows container. Currently windows containers < 1809 do not support symlinks well, so disabling symlink resolution/dedupe behavior");
-            return true;
-        }
-        else
-        {
-            return false;
-        }
-    }
+    public string ResolvePhysicalPath(string path) => Path.GetFullPath(path);
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Common/PathUtilityService.cs:65]: Does `GetFullPath` follow symlinks?

**Review by JamieMagee — APPROVED** [review]
Approved pending resolving comments.

File: src/Microsoft.ComponentDetection.Common/PathUtilityService.cs (hunk 1 above)
Comment (melotic) [reply src/Microsoft.ComponentDetection.Common/PathUtilityService.cs:65]: Yes: 
image

Comment (melotic) [reply src/Microsoft.ComponentDetection.Common/PathUtilityService.cs]: It does not. I've updated it to properly create either a `FileInfo` or `DirectoryInfo` class, then we can use the [`LinkTarget`](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.linktarget?view=net-7.0) property.

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**